src/parallel/task_scheduler.cpp

```diff
@@ -124,15 +125,25 @@ bool TaskScheduler::GetTaskFromProducer(ProducerToken &token, shared_ptr<Task> &
 	return queue->DequeueFromProducer(token, task);
 }
 
+static inline int64_t CurrentTimeMS() {
+	return duration_cast<std::chrono::milliseconds>(system_clock::now().time_since_epoch()).count();
+}
+
 void TaskScheduler::ExecuteForever(atomic<bool> *marker) {
 #ifndef DUCKDB_NO_THREADS
+	constexpr static int64_t TASK_DURATION_FLUSH_THRESHOLD_MS = 100;
```

Comment:
Perhaps this should be a configuration parameter - with also an optional way to disable it?